lib/logstash/inputs/lumberjack.rb

```diff
@@ -58,7 +54,6 @@ def register
     # start rejecting new connection and raise an exception
     @threadpool = Concurrent::ThreadPoolExecutor.new(
       :min_threads => 1,
-      :max_threads => @max_clients,
```

Comment:
There's no max, to my knowledge. Given the current work model of the ruby lumberjack input, no maximum is fine.

The intended behavior of the circuit breaker is that we stop accepting _new_ connections if logstash gets wedged, and resume accepting when logstash is happy.

Because each connection is handled in a new thread (per connection), and we don't want  to limit the number of connections, it makes sense to remove any ceiling of max connections. The max connections setting was only added as as temporary solution to the lumberjack-retransmit-forever-causing-oom problem until we had a better (this circtuibreaker) solution.